kata-monitor: add the README file

fgiudici opened the pull request:
The documentation we already had was in the metrics 2.0 design document.
Added the kata-monitor README, trying to not duplicate info (or at least not too much) and linking back the the metrics 2.0 design document.
Fixes: #3704

fidencio commented:
Can one of the admins verify this patch?

cmaf commented on src/runtime/cmd/kata-monitor/README.md:
```diff
@@ -0,0 +1,59 @@
+
+**NOTE: The debug endpoints are available only if the [Kata Containers configuration file](../../config/configuration-qemu.toml.in#L590-L592) includes** `enable_pprof = true` **in the** `[runtime]` **section**.
+
+The `/sandboxes` endpoint lists the sandbox id of all the detected Kata runtimes. If accessed via a web browser, it provides html links to the endpoints available for each sandbox.
```
s/id/ID unless it is a variable name, then make some indication such as you do below

fgiudici replied:
fixed, thanks!

cmaf commented on src/runtime/cmd/kata-monitor/README.md:
```diff
@@ -0,0 +1,59 @@
+  * Firecracker metrics
+  * Kata monitor metrics
+
+All the provided metrics are in Prometheus format. While Kata monitor can be used as a standalone daemon on any host running Kata Containers workloads and can be used for retrieving profiling data from the running Kata runtimes, its main expected usage is to be deployed as a Daemonset on a Kubernetes cluster: there Prometheus should scrape the metrics from the kata-monitor endpoints.
```
Kata monitor should maybe be kata-monitor here since it refers to the daemon

fgiudici replied:
Right, makes sense, changed

cmaf commented on src/runtime/cmd/kata-monitor/README.md:
```diff
@@ -0,0 +1,59 @@
+For more information on the Kata Containers metrics architecture and a detailed list of the available metrics provided by Kata monitor check the [Kata 2.0 Metrics Design](../../../../docs/design/kata-2-0-metrics.md) document.
+
+## Usage
+Each Kata monitor instance detects and monitors the Kata Container workloads running on the same node.
```
also possibly kata-monitor here

fgiudici replied:
done!

cmaf reviewed (commented):
thanks @fgiudici, a few nits

fgiudici commented:
@cmaf , thanks for the review!
Apart from the fixes you suggested I added a couple more to adhere to the Doc guidelines:
* changed the final "Example" section, splitting commands from outputs
* added a link to the kata-monitor README in the most close README ( i.e., the one under src/runtime)

fgiudici commented:
/test

liubin commented on src/runtime/cmd/kata-monitor/README.md:
```diff
@@ -0,0 +1,68 @@
+
+* `--listen-address` _IP:PORT_
+* `--runtime-enpoint` _PATH_TO_THE_CONTAINER_MANAGER_CRI_INTERFACE_
+* `--log-level` {trace|debug|info|warn|error|fatal|panic}
```
Is it better change this line to:

```
* `--log-level` _[trace|debug|info|warn|error|fatal|panic]_
```
Then all three items will have the same style of italic.

fgiudici replied:
fixed, thanks :-)

cmaf reviewed (approved):
thanks @fgiudici

liubin commented:
/test
